### PR TITLE
WIP/RFC add some introductory words/links to doc frontpage

### PR DESCRIPTION
**Pull request by KristofferC**
The current front page of the documentation is a bit bare in my opinion.

This adds a welcome message, adds the NEWS document to the docs with a link and a list of blogs (only 1 so far) relevant for 0.6.
Example at https://kristofferc.github.io/JuliaDocFrontPageEx/.

Links to issues and PRs in `NEWS.md` are currently not parsed correctly.

Please comment.

### CODE REVIEW

**Comment by dpsanders** [comment]
That's a good start.

- I suggest `What is new` -> `What's new`. 

- There is some inconsistency about `0.6` vs `0.6.0`. I suggest using `0.6`

- It would be good to add a few words about the difference between the `Manual` and `Standard Library` sections (which, to be honest, still confuses me); should `Standard Library` in fact be renamed to `Standard library reference`?

File: doc/src/index.md [hunk 1]
```diff
@@ -1,4 +1,15 @@
-# Julia Documentation
+# Julia v0.6 Documentation
+
+Welcome to the documentation for Julia 0.6!
+
+### What's new in Julia 0.6
+
+Please see the [Release notes](NEWS.md) for a list of new features, changes and deprecations from 0.5 to 0.6.
+
+For more in detail analysis of some of the new functionalities in 0.6, see the following blog posts:
```
Comment (Sacha0) [inline doc/src/index.md]: Perhaps the more colloquial "For a closer look at some of 0.6's new features, see..."?

Comment (ararslan) [inline doc/src/index.md]: Hmm, I think I actually prefer how it was before this change. "0.6's" reads kind of weird to me.

Comment (KristofferC) [inline doc/src/index.md]: I agree with `@Sacha0` that the first one was a bit too formal.

Comment (ararslan) [inline doc/src/index.md]: Okay. What you have now is fine anyway, IMO. Nice work here!

Comment (Sacha0) [inline doc/src/index.md]: Avoiding "0.6's", perhaps "For a closer look at some of this release's new features, see..." or "For a closer look at [certain|specific|particular] new features in 0.6, see..."?

**Comment by dpsanders** [comment]
How about something like "For an overview of new functionality in 0.6, see..."

**Comment by KristofferC** [comment]
I made it so that the links in the NEWS file work (by dynamically creating a fixed version from the base NEWS.md) file. Also added two more blog posts.

**Comment by fredrikekre** [comment]
Could you update https://kristofferc.github.io/JuliaDocFrontPageEx/?

**Comment by KristofferC** [comment]
Update it.

File: doc/src/index.md [hunk 2]
```diff
@@ -1,4 +1,15 @@
-# Julia Documentation
+# Julia v0.6 Documentation
+
+Welcome to the documentation for Julia 0.6!
+
+### What's new in Julia 0.6
+
+The [Release notes](NEWS.md) contains a list of new features, changes and deprecations.
```
Comment (Sacha0) [inline doc/src/index.md]: "Release" -> "release", and "contains" -> "contain"? Perhaps simplify to "The release notes list new features, changes, and deprecations."?

File: doc/src/index.md [hunk 3]
```diff
@@ -1,4 +1,15 @@
-# Julia Documentation
+# Julia v0.6 Documentation
+
+Welcome to the documentation for Julia 0.6!
+
+### What's new in Julia 0.6
+
+The [release notes](NEWS.md) contain a list of new features, changes and deprecations.
+A closer look at some of the larger, new, functionalities can be found in the following blog posts:
```
Comment (ararslan) [inline doc/src/index.md]: No comma after "new", but I would reword as "A closer look at some of the biggest changes and improvements can be found in the following blog posts:"

File: doc/make.jl [hunk 4]
```diff
@@ -14,10 +14,14 @@ isdefined(:build_sysimg) || @eval module BuildSysImg
     include(joinpath(@__DIR__, "..", "contrib", "build_sysimg.jl"))
 end
 
+include("NEWS-makedoc.jl")
```
Comment (fredrikekre) [inline doc/make.jl]: Forgot to add this file?
